### PR TITLE
Xcm comissions

### LINKED ISSUE #119: [Tech]: XCM Fee Management
### Problem statement

- For each **incoming** XCM message the parachain takes some fees on assets and burns some of them (value in units needed to execute transaction)
- Since the parachain has not any tokenomics there are two ways to organize the transaction fee withdrawal
1) Calculate the rate of each asset on the parachain and take some fees based on that. The fees would be sent to outbound channel to technical account or other treasury account or an entity
2) Calculate the rate of each asset on Sora network.

The second way is more relevant because it is easier to organize the asset rate calculation on sora side. Bit the biggest drawback of that way - is to leave parachain without transaction fees that leads to security vulnerability. Thus, parachain should have a protection from spamming with low amounts of tokens - so for each registered asset the minimum amount that can be passed through the bridge should exist.

### Description

_No response_

### Definition of Done

1. Create a storage that keeps the information about minimum amount of assets that can be sent to it via XCM
2. Create a code that checks for this amount and returns an error for each amount
3. Disable fee collection for incoming XCM messages

### Requirements

_No response_

### PR DESCRIPTION
**Pull request by N1ghtStorm**

### CODE REVIEW

File: pallets/xcm-app/src/lib.rs [hunk 1]
```diff
@@ -358,6 +370,22 @@ pub mod pallet {
...
+        pub fn set_asset_minimum_amount(
+            origin: OriginFor<T>,
+            asset_id: AssetId,
+            min_amount: u128,
+        ) -> DispatchResultWithPostInfo {
+            let _ = T::CallOrigin::ensure_origin(origin)?;
+            let Some(multilocation) = Self::get_multilocation_from_asset_id(asset_id) else {
+                fail!(Error::<T>::MappingNotExist);
+            };
+            AssetMinimumAmount::<T>::set(multilocation, Some(min_amount));
```
Comment (wer1st) [inline pallets/xcm-app/src/lib.rs]: What about using of this value? Will you have another PR for that?

File: pallets/xcm-app/src/lib.rs [hunk 2]
```diff
@@ -275,6 +285,7 @@ pub mod pallet {
             asset_id: AssetId,
             multiasset: xcm::v3::AssetId,
             asset_kind: bridge_types::types::AssetKind,
+            min_amount: u128,
```
Comment (wer1st) [inline pallets/xcm-app/src/lib.rs]: Do you have special type for balance here?

Comment (N1ghtStorm) [reply pallets/xcm-app/src/lib.rs]: no, the belance is mapped properly on sora2-network

File: pallets/xcm-app/src/lib.rs (hunk 1 above)
Comment (N1ghtStorm) [reply pallets/xcm-app/src/lib.rs]: you can see usage in this pr in runtime/src/trader

File: Cargo.toml [hunk 3]
```diff
@@ -2,32 +2,32 @@
...
-leaf-provider-rpc = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-leaf-provider-runtime-api = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-bridge-common = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-dispatch = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-bridge-types = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-bridge-data-signer = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-multisig-verifier = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-substrate-bridge-channel = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
-bridge-channel-rpc = { git = "https://github.com/sora-xor//sora2-common.git", tag = "1.0.8" }
+beefy-light-client = { path = "../sora2-common/pallets/beefy-light-client" }
```
Comment (Alexey-N-Chernyshov) [inline Cargo.toml]: please, rollback this file

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: Cargo.toml (hunk 3 above)
Comment (N1ghtStorm) [reply Cargo.toml]: > please, rollback this file

shure, I will change it after merging this https://github.com/sora-xor/sora2-common/pull/50 and creating tag

**Pull request merged**